webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -82,6 +88,16 @@ export class ContactsEffects {
             console.error('Error selecting contact', err);
             this.globalActions.unsetSelected();
             return of(this.contactsActions.clearSelection());
+          })
+          .finally(() => {
+            trackPerformance?.stop({
+              recordApdex: true,
+              name: [
+                'select_contact',
+                selectedContact?.doc?.contact_type || 'contact',
```

Comment:
Would it be easier to get the selectedContact from the store rather than in a variable?

Reply:
I'm afraid of a race condition or retriggering this effect and causing more spaghetti in the execution flow. 
This is not the nicest, but it's safe and simple to debug. Ideally, we refactor all this code to simplify, but not in this PR scope.